lib/streamlit/DeltaGenerator.py

```diff
@@ -2505,16 +2505,17 @@ def progress(self, element, value):
         ...     my_bar.progress(percent_complete + 1)
 
         """
-        # Needed for python 2/3 compatibility
-        value_type = type(value).__name__
-        if value_type == "float":
+        int_value = isinstance(value, numbers.Integral)
```

Comment:
I'd suggest just inlining these in the conditions; I think it would be easier to read.